runtime/src/main/java/org/corfudb/runtime/CorfuRuntime.java

```diff
@@ -659,12 +613,199 @@ public synchronized CorfuRuntime connect() {
             fastLoader.loadMaps();
         }
         return this;
     }
+
+    // Below are deprecated methods which should no longer be
+    // used and may be deprecated in the future.
+
+    // region Deprecated Constructors
+    /**
+     * Constructor for CorfuRuntime.
+     * @deprecated Use {@link CorfuRuntime#fromParameters(CorfuRuntimeParameters)}
+     **/
+    @Deprecated
+    public CorfuRuntime() {
+        this(CorfuRuntimeParameters.builder().build());
+    }
+
+    /**
+     * Parse a configuration string and get a CorfuRuntime.
+     *
+     * @param configurationString The configuration string to parse.
+     * @deprecated Use {@link CorfuRuntime#fromParameters(CorfuRuntimeParameters)}
+     */
+    @Deprecated
+    public CorfuRuntime(String configurationString) {
+        this(CorfuRuntimeParameters.builder().build());
+        this.parseConfigurationString(configurationString);
+    }
+    // endregion
+
+    // region Deprecated Setters
+
+    /**
+     * Enable TLS.
+     * @deprecated  Deprecated, set using {@link CorfuRuntimeParameters} instead.
+     **/
+    @Deprecated
+    public CorfuRuntime enableTls(String keyStore, String ksPasswordFile, String trustStore,
+        String tsPasswordFile) {
+        log.warn("enableTls: Deprecated, please set parameters instead");
+        parameters.keyStore = keyStore;
+        parameters.ksPasswordFile = ksPasswordFile;
+        parameters.trustStore = trustStore;
+        parameters.tsPasswordFile = tsPasswordFile;
+        parameters.tlsEnabled = true;
+        return this;
+    }
+
+    /**
+     * Enable SASL Plain Text.
+     * @deprecated  Deprecated, set using {@link CorfuRuntimeParameters} instead.
+     **/
+    @Deprecated
+    public CorfuRuntime enableSaslPlainText(String usernameFile, String passwordFile) {
+        log.warn("enableSaslPlainText: Deprecated, please set parameters instead");
+        parameters.usernameFile = usernameFile;
+        parameters.passwordFile = passwordFile;
+        parameters.saslPlainTextEnabled = true;
+        return this;
+    }
+
+    /**
+     * Whether or not to disable backpointers
+     *
+     * @param disable True, if the cache should be disabled, false otherwise.
+     * @return A CorfuRuntime to support chaining.
+     * @deprecated  Deprecated, set using {@link CorfuRuntimeParameters} instead.
+     */
+    @Deprecated
+    public CorfuRuntime setBackpointersDisabled(boolean disable) {
+        log.warn("setBackpointersDisabled: Deprecated, please set parameters instead");
+        parameters.setBackpointersDisabled(disable);
+        return this;
+    }
+
+    /**
+     * Whether or not to disable the cache
+     *
+     * @param disable True, if the cache should be disabled, false otherwise.
+     * @return A CorfuRuntime to support chaining.
+     * @deprecated  Deprecated, set using {@link CorfuRuntimeParameters} instead.
+     */
+    @Deprecated
+    public CorfuRuntime setCacheDisabled(boolean disable) {
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Do not forget to remove this deprecated code someday. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1133)